CotEditor/Sources/Document.swift

```diff
@@ -337,6 +337,8 @@ final class Document: NSDocument, AdditionalDocumentPreparing, EncodingHolder {
             ?? Defaults[.syntaxStyle]
         
         self.setSyntaxStyle(name: styleName)
+        
+        ScriptManager.shared.dispatchEvent(documentOpened: self)
```

Comment:
`read(from:, ofType:)` is invoked not only on file opening but also on file reverting (namely, both on "Revert To..." action and on resolving a file update conflict). 

Thus, I suggest not to call `ScriptManager.shared.dispatchEvent(documentOpened: self)` here but somewhere else. Well..., I'd say `didMakeDocumentForExisitingFile(url:)` is better. The method is invoked after `read(from:, ofType:)` and before document-related views are loaded.